src/agent/services/oracle.rs

```diff
@@ -129,14 +156,17 @@ where
             Some(account) => {
                 let pubkey: Pubkey = update.value.pubkey.as_str().try_into()?;
                 let state = state.clone();
-                tokio::spawn(async move {
-                    if let Err(err) =
-                        Oracle::handle_price_account_update(&*state, network, &pubkey, &account)
-                            .await
-                    {
-                        tracing::error!(err = ?err, "Failed to handle account update.");
-                    }
-                });
+                sender
+                    .send(tokio::spawn(async move {
+                        if let Err(err) =
+                            Oracle::handle_price_account_update(&*state, network, &pubkey, &account)
```

Comment:
I wonder why it's possible to accumulate a lot of long-lived tasks. Maybe we should add a timeout to `Oracle::handle_price_account_update`? Especially considering that now if tasks become stuck, handling price updates can halt completely.

Reply:
@Riateche same here